### PR TITLE
use the StringBuilder another method can reduce impact on array copy performance

### PR DESCRIPTION
**Pull request by gylang**
…performance

use the StringBuilder another method can reduce impact on array copy performance

**Make sure set the target branch to `develop`**

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by odbozhou** [comment]
Please submit an issue first, this pr is associated with this issue

File: store/src/main/java/org/apache/rocketmq/store/index/IndexFile.java [hunk 1]
```diff
@@ -71,6 +72,9 @@ public void load() {
         this.indexHeader.load();
     }
 
+    /**
+     * 头部文件刷盘 时间戳 msgId 物理偏移量..
```
Comment (AbelSara) [inline store/src/main/java/org/apache/rocketmq/store/index/IndexFile.java:76]: Why do you use so many annotations, and it is in Chinese?

Comment (gylang) [reply store/src/main/java/org/apache/rocketmq/store/index/IndexFile.java:76]: sory, I don't know why this code was submitted here

Comment (gylang) [reply store/src/main/java/org/apache/rocketmq/store/index/IndexFile.java:76]: this code just my study comment. i do'not want to pr the "3eee57d"

Comment (gylang) [reply store/src/main/java/org/apache/rocketmq/store/index/IndexFile.java:76]: this code just my study comment. i don't want to pr "3eee57d"

**Comment by areyouok** [comment]
the develop branch has force-pushed. please rebase this pr.

git rebase develop

**Review by caigy — CHANGES_REQUESTED** [review]
Pls remove unrelated modifications.

**Pull request closed without merging**